projects/max14906/src/examples/dummy/dummy_example.c

```diff
@@ -0,0 +1,150 @@
+#include "dummy_example.h"
+#include "common_data.h"
+#include "max149x6-base.h"
+#include "max14906.h"
+#include "no_os_delay.h"
+#include "no_os_print_log.h"
+
+void gpio_callback_fn(void *ctx)
+{
+	struct max149x6_desc *desc = ctx;
+	int ret;
+	uint32_t val;
+
+	ret = max149x6_reg_read(desc, MAX14906_INT_REG, &val);
+	if (val)
+		pr_info("Fault detected!\n");
+}
+
+int dummy_example_main()
+{
+	int ret, i;
+	uint32_t val;
+	struct max149x6_desc *max14906_desc;
+
+	enum max14906_climit climit = MAX14906_CL_300;
+	enum max14906_climit climit2 = MAX14906_CL_600;
+
+	struct no_os_gpio_desc *gpio_desc;
+
+	/** MAX14906 Initialization */
+	ret = max14906_init(&max14906_desc, &max14906_ip);
+	if (ret)
+		goto error;
+
+	/** Global Interrupt Controller */
+	struct no_os_irq_ctrl_desc *global_desc;
+	struct no_os_irq_init_param global_desc_param = {
+		.irq_ctrl_id = GPIO_IRQ_ID,
+		.platform_ops = &max_irq_ops,
+		.extra = NULL
+	};
+
+	ret = no_os_irq_ctrl_init(&global_desc, &global_desc_param);
+	if (ret)
+		return ret;
```

Comment:
make sure all the remove APIs are called in case of an error to avoid memory leaks, apply to all the returns in this function.

Reply:
I am verifying this.